jina/types/document/__init__.py

```diff
@@ -534,3 +534,24 @@ def MergeFrom(self, doc: 'Document'):
 
     def CopyFrom(self, doc: 'Document'):
         self._document.CopyFrom(doc.as_pb_object)
+
+    def traverse_apply(self, traversal_paths: Tuple[str], apply_func: Callable, *args, **kwargs) -> None:
+        """Traverse leaves of the document."""
+        for path in traversal_paths:
+            if path[0] == 'r':
+                self._traverse_rec(self, None, None, [], apply_func, *args, **kwargs)
+            self._traverse_rec(self, None, None, path, apply_func, *args, **kwargs)
+
+    def _traverse_rec(self, parent_doc, parent_edge_type, path, apply_func, *args, **kwargs):
+        if path:
+            next_edge = path[0]
+            if next_edge == 'm':
+                self._traverse_rec(
```

Comment:
Just a comment, This call does not match signature right?

Reply:
yes, I'm working on that